app/views/layouts/application.html.erb

```diff
@@ -10,6 +10,39 @@
   </head>
 
   <body onload="loadLocation();">
+    <script>
+      window.fbAsyncInit = function() {
+        FB.init({
+          appId      : '1653169284778740',
+          cookie     : true,
+          xfbml      : true,
+          version    : 'v2.12'
+        });
+
+        FB.AppEvents.logPageView();
+
+      };
+
+      (function(d, s, id){
+         var js, fjs = d.getElementsByTagName(s)[0];
+         if (d.getElementById(id)) {return;}
+         js = d.createElement(s); js.id = id;
+         js.src = "https://connect.facebook.net/en_US/sdk.js";
+         fjs.parentNode.insertBefore(js, fjs);
+       }(document, 'script', 'facebook-jssdk'));
+    </script>
+
+    <div class='fb_block'>
```

Comment:
Move this to a partial too, that way the layout main file is more clear and concise